packages/ses/src/module-load.js

```diff
@@ -0,0 +1,102 @@
+const { create, keys, values, freeze } = Object;
+
+// makeAlias constructs compartment specifier tuples for the `aliases` private
+// field of compartments.
+// These aliases allow a compartment to alias an internal module specifier to a
+// module specifier in an external compartment, and also to create internal
+// aliases.
+// Both are facilitated by the moduleMap Compartment constructor option.
+export const makeAlias = (compartment, specifier) =>
+  freeze({
+    compartment,
+    specifier,
+  });
+
+// Pre-compute resolutions of all imports within the compartment in which a
+// module was loaded.
+const resolveAll = (imports, resolveHook, fullReferrerSpecifier) => {
+  const resolvedImports = create(null);
+  for (const importSpecifier of keys(imports)) {
+    const fullSpecifier = resolveHook(importSpecifier, fullReferrerSpecifier);
+    resolvedImports[importSpecifier] = fullSpecifier;
+  }
+  return freeze(resolvedImports);
+};
+
+// load asynchronously loads ModuleStaticRecords and creates a complete graph
```

Comment:
Did you mean "load asynchronously loads ModuleStaticRecords ..." or
"Asynchronously loads ModuleStaticRecords ..."
?

Reply:
I’ve gotten in the habit, from my time with the Go linter, beginning doc comments with the subject, so I did intentionally stutter “load” here, noun then verb. But, I can see it’s jarring. I’ve poked about for precedent, but I’m not sure there is one. Glad for any recommendation.

Reply:
Ah. Having been habituated by markdown, I would have written this as

```js
// `load` asynchronously loads `ModuleRecords` ...
```

However, I have no idea whether this confuses or is understood by any of the tooling around jsdoc comments. In any case, without some marking, I had missed that the first word was the function name. Ignore my earlier suggestion.

Reply:
I like the Markdown convention.